src/lib/object.test.ts

```diff
@@ -53,4 +53,85 @@ describe('clone', () => {
     const cloned = clone(object);
     expect(cloned).toStrictEqual(cloned);
   });
+
+  it('should clone undefined', () => {
+    const object = undefined;
+    const cloned = clone(object);
+    expect(cloned).toStrictEqual(undefined);
+  });
+
+  it('should clone null', () => {
+    const object = null;
+    const cloned = clone(object);
+    expect(cloned).toStrictEqual(null);
+  });
+
+  it('should clone empty object', () => {
+    const object = {};
+    const cloned = clone(object);
+    expect(cloned).toStrictEqual({});
+  });
+
+  describe('when cloning buffer', () => {
+    let object: { buffer: Buffer };
+
+    beforeEach(() => {
+      object = {
+        buffer: Buffer.from('data'),
+      };
+    });
+
+    it('should clone buffer', () => {
+      const cloned = clone(object);
+      expect(Buffer.isBuffer(cloned.buffer)).toBe(true);
+      expect(cloned.buffer.toString()).toEqual(object.buffer.toString());
+    });
+
+    it('should create new instance of buffer', () => {
+      const cloned = clone(object);
+      expect(cloned.buffer !== object.buffer).toBe(true);
+    });
+  });
+
+  describe('when cloning array', () => {
+    let object: { array: Array<any> };
+
+    beforeEach(() => {
+      object = {
+        array: [1, { a: 1, b: 2, c: 'string' }],
+      };
+    });
+
+    it('should clone array', () => {
+      const cloned = clone(object);
+      expect(Array.isArray(cloned.array)).toBe(true);
+      expect(cloned).toStrictEqual(object);
+    });
+
+    it('should create new instance of array', () => {
+      const cloned = clone(object);
+      expect(cloned.array !== object.array).toBe(true);
```

Comment:
I recommend using Jest's built-in expectation for checking referential inequality, it's shorter and more readable:

```suggestion
      expect(cloned).not.toBe(object);
```

Alternatively, since we don't care about deep comparison here, per [docs](https://jestjs.io/docs/expect#tobevalue) you may want to use `Object.is` directly (which is preferred over `===`):

```suggestion
      expect(Object.is(object, cloned)).toBe(false)
```

Same applies to lines 92 and 134.